ocaml-lsp-server/src/inferred_intf.ml

```diff
@@ -0,0 +1,85 @@
+open Import
+
+let action_kind = "inferred_intf"
+
+let infer_intf impl =
+  Document.with_pipeline impl (fun pipeline ->
+      let typer = Mpipeline.typer_result pipeline in
+      let pos = Mpipeline.get_lexing_pos pipeline `Start in
+      let env, _ = Mbrowse.leaf_node (Mtyper.node_at typer pos) in
+      let sig_ : Types.signature =
+        let typedtree = Mtyper.get_typedtree typer in
+        match typedtree with
+        | `Interface _ -> assert false
+        | `Implementation impl -> impl.str_type
+      in
+      Printtyp.wrap_printing_env env (fun () ->
+          Format.asprintf "%a@." Printtyp.signature sig_))
+
+let code_action_of_intf uri intf range =
+  let edit : WorkspaceEdit.t =
+    let textedit : TextEdit.t = { range; newText = intf } in
+    let uri = Uri.to_string uri in
+    WorkspaceEdit.create ~changes:[ (uri, [ textedit ]) ] ()
+  in
+  let title = String.capitalize_ascii "Insert inferred interface" in
+  CodeAction.create ~title ~kind:(CodeActionKind.Other action_kind) ~edit
+    ~isPreferred:false ()
+
+let read_file filename =
+  let ch = open_in filename in
+  let s = really_input_string ch (in_channel_length ch) in
+  close_in ch;
+  s
```

Comment:
you could do `Io.read_file`, which is contained in dune's stdlib `Stdune` that we use and have opened in `Import`

Reply:
It does not seem to be in `Import`, but I can add it if that makes sense

Reply:
Yes, please add it. Feel free to use whatever modules you want from stdune. The only reason why we don't do `open Stdune` is because some modules shadow modules from compiler-libs (`Env` for example).

Reply:
I added it 👍